lib/Logger/Logger.cpp

```diff
@@ -553,21 +614,28 @@ void Logger::log(char const* logid, char const* function, char const* file,
       dumper.appendString(_hostname.data(), _hostname.size());
     }
 
-    // meta data from log context
-    LogContext::OverloadVisitor visitor(
-        [&out, &dumper](std::string_view const& key, auto&& value) {
-          out.push_back(',');
-          dumper.appendString(key.data(), key.size());
-          out.push_back(':');
-          if constexpr (std::is_same_v<std::string_view,
-                                       std::remove_cv_t<std::remove_reference_t<
-                                           decltype(value)>>>) {
-            dumper.appendString(value.data(), key.size());
-          } else {
-            out.append(std::to_string(value));
-          }
-        });
-    logContext.visit(visitor);
+    {
+      READ_LOCKER(guard, _structuredParamsLock);
+      // meta data from log context
+      LogContext::OverloadVisitor visitor([&out, &dumper](
+                                              std::string_view const& key,
+                                              auto&& value) {
+        if (!_structuredLogParams.contains(key.data())) {
```

Comment:
Not sure if `key` is always null-terminated here. Can we also pass the size of `key` when doing the lookup?
```suggestion
        if (!_structuredLogParams.contains(key.data())) {
```

Reply:
Addressed